### PR TITLE
add the THPRESFT keyword

### PR DESCRIPTION
**Pull request by andlaus**
it is like THPRES, but for faults instead of EQUIL regions.

### CODE REVIEW

**Comment by andlaus** [comment]
jenkins build this please

**Comment by andlaus** [comment]
jenkins build this please

**Comment by andlaus** [comment]
With jenkins now happy, I think that this is merge-ready.

File: src/opm/parser/eclipse/EclipseState/Grid/SatfuncPropertyInitializers.cpp [hunk 1]
```diff
@@ -55,8 +55,12 @@ namespace Opm {
         const TableContainer& sgfnTables = tm->getSgfnTables();
 
 
-        bool family1 = (!sgofTables.empty() || !slgofTables.empty()) && !swofTables.empty();
-        bool family2 = !swfnTables.empty() && !sgfnTables.empty() && !sof3Tables.empty();
+        bool family1 = !sgofTables.empty() || !swofTables.empty() || !slgofTables.empty();
+        bool family2 = !swfnTables.empty() || !sgfnTables.empty() || !sof3Tables.empty();
+
+        if (!swofTables.empty() && !slgofTables.empty()) {
```
Comment (joakim-hove) [inline src/opm/parser/eclipse/EclipseState/Grid/SatfuncPropertyInitializers.cpp]: The diff in `SatfuncPropertyInitializers.cpp` seems unrelated to `THPRESFT`- PR pollution?

**Comment by joakim-hove** [comment]
I had one minor comment about a diff which propably comes from #692 - otherwise this looks innocent enough and ready to merge; however it does not seem like the current PR changes anything - i.e. the newly introduced `THPRESFT` keyword is not used for anything - or??

File: src/opm/parser/eclipse/EclipseState/Grid/SatfuncPropertyInitializers.cpp (hunk 1 above)
Comment (andlaus) [reply src/opm/parser/eclipse/EclipseState/Grid/SatfuncPropertyInitializers.cpp]: that was unintentional, yes. shall I split it into a separate patch and leave it in on PR, do you want a separate PR, or shall it stay like it is in order to reduce the number of emails in people's inboxes?

**Comment by andlaus** [comment]
> however it does not seem like the current PR changes anything - i.e. the newly introduced THPRESFT keyword is not used for anything - or??

in one sense, that is correct: the threshold pressures are not modified by this in itself. (this requires simulator support. patch coming.)  in another sense there is a change: the THPRES option for the EQLOPTS keyword now gets accepted even if only THPRESFT is present, but maybe this option should simply always be accepted?!

File: src/opm/parser/eclipse/EclipseState/Grid/SatfuncPropertyInitializers.cpp (hunk 1 above)
Comment (joakim-hove) [reply src/opm/parser/eclipse/EclipseState/Grid/SatfuncPropertyInitializers.cpp]: Assuming the offending changes should have been part of #692 I would like you to fixup that and we can merge that out of the way.

**Comment by joakim-hove** [comment]
> in one sense, that is correct: the threshold pressures are not modified by this in itself. (this requires simulator support. patch coming.)

OK - but looking at the existing `THPRES` support there is [small class](https://github.com/OPM/opm-common/blob/master/src/opm/parser/eclipse/EclipseState/SimulationConfig/ThresholdPressure.cpp) to actually implement the THPRES behavior - I would assume something similar was needed also for `THPRESFT`?  So although a patch is needed in opm-simulators, I would say more work was needed also here in opm-common?

> in another sense there is a change: the THPRES option for the EQLOPTS keyword now gets accepted even if only THPRESFT is present,

Well - that acceptance is a little premature, since only the presence of the `THPRESFT` keyword is acknowledged, the actual content is fully ignored?

So - unless I misunderstand big time, I would say this PR needs more functionality before it is interesting to merge?

File: src/opm/parser/eclipse/EclipseState/Grid/SatfuncPropertyInitializers.cpp (hunk 1 above)
Comment (andlaus) [reply src/opm/parser/eclipse/EclipseState/Grid/SatfuncPropertyInitializers.cpp]: okay, I factored this out into a separate patch and moved it to #692.

**Comment by andlaus** [comment]
jenkins build this please

**Comment by andlaus** [comment]
> Well - that acceptance is a little premature, since only the presence of the THPRESFT keyword is acknowledged, the actual content is fully ignored?

I agree that this does not make much sense, but the eclipse RM documentaion for EQLOPTS says that if you want to use THPRESFT, you have to specify the THPRES option in EQLOPTS.

**Comment by andlaus** [comment]
IOW, it would be sufficient not to touch that class if the creation of the deck does not fail anymore if only the THPRESFT keyword is present and the THPRES option in EQLOPTS is set (which would also be akward, because there *are* threshold pressures then).

**Comment by joakim-hove** [comment]
To be honest I still do not really understand this - but PR itself seems innocent enough - and correct. Merging.
